tests/e2e/conftest.py

```diff
@@ -25,6 +31,163 @@
+        awscli_pod_session=awscli_pod_session,
+        mcg_obj_session=mcg_obj_session,
+    ):
+        # create bucket and write some objects to it
+        test_bucket = bucket_factory()[0]
+        write_random_test_objects_to_bucket(
+            io_pod=awscli_pod_session,
+            file_dir="test_dir",
+            pattern="test-object",
+            bucket_to_write=test_bucket.name,
+            mcg_obj=mcg_obj_session,
+        )
+
+        # Backup secrets to local folders
+        ocp_secret_obj = OCP(
+            kind="secret", namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        secrets = [
+            "noobaa-root-master-key",
+            "noobaa-admin",
+            "noobaa-operator",
+            "noobaa-db",
+            "noobaa-server",
+            "noobaa-endpoints",
+        ]
+        secrets_yaml = [
+            ocp_secret_obj.get(resource_name=f"{secret}") for secret in secrets
+        ]
+        secrets_obj = [OCS(**secret_yaml) for secret_yaml in secrets_yaml]
+        logger.info(secrets_obj)
+        logger.info("Backed up secrets to local folders!")
+
+        # Backup the PostgreSQL database and save it to a local folder
+        noobaa_db_pod = Pod(
+            **get_pods_having_label(
+                label="noobaa-db=postgres",
+                namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            )[0]
+        )
+        noobaa_db_pod.exec_cmd_on_pod(
+            command="pg_dump nbcore -f /tmp/test.db -F custom"
+        )
+        OCP(namespace=constants.OPENSHIFT_STORAGE_NAMESPACE).exec_oc_cmd(
+            command=f"cp {noobaa_db_pod.name}:/tmp/test.db ./mcg.bck"
+        )
+        logger.info("Backed up PostgreSQL and stored it in local folder!")
+
+        # For testing, write new data to show a little data loss between backup and restore
+        testloss_bucket = bucket_factory()[0]
+        write_random_test_objects_to_bucket(
+            io_pod=awscli_pod_session,
+            file_dir="testloss_dir",
+            pattern="testloss-object",
+            bucket_to_write=testloss_bucket.name,
+            mcg_obj=mcg_obj_session,
+        )
+
+        # Stop MCG reconcilation
+        ocp_storagecluster_obj = OCP(
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE, kind="storagecluster"
+        )
+        params = '{"spec": {"multiCloudGateway": {"reconcileStrategy": "ignore"}}}'
+        ocp_storagecluster_obj.patch(
+            resource_name="ocs-storagecluster", params=params, format_type="merge"
+        )
+        logger.info("Stopped MCG reconcilation!")
+
+        # Stop the NooBaa Service before restoring the NooBaa DB. There will be no object service after this point
+        ocp_deployment_obj = OCP(
+            kind="deployment", namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        nb_operator_dc = Deployment(
+            **ocp_deployment_obj.get(resource_name="noobaa-operator")
+        )
+        nb_operator_dc.scale(replicas=0)
```

Comment:
why not modify_deployment_replica_count?

Reply:
`modify_deployment_replica_count` creates unnecessary overhead of creating ocp object everytime function is called. Also having deployment object re usability makes in this scenario with simple scale method